includes/admin/extensions/class-extensions-api.php

```diff
@@ -0,0 +1,222 @@
+		);
+		$request = wp_remote_get(
+			$url,
+			array(
+				'timeout'   => 15,
+				'sslverify' => true,
+			)
+		);
+
+		// If there was an API error, set option and return false.
+		if ( is_wp_error( $request ) || ( 200 !== wp_remote_retrieve_response_code( $request ) ) ) {
+			update_option(
+				'edd_all_extension_data',
+				array(
+					'timeout' => strtotime( '+1 hour', time() ),
+				),
+				false
+			);
+
+			return false;
+		}
+
+		// Fresh data has been retrieved, so update the option with a four hour timeout.
+		$all_product_data = json_decode( wp_remote_retrieve_body( $request ) );
+		$data             = array(
+			'timeout'  => strtotime( '+4 hours', time() ),
+			'products' => $all_product_data,
+		);
+
+		update_option( 'edd_all_extension_data', $data, false );
+
+		return $all_product_data;
+	}
+
+	/**
+	 * Gets the product data as needed for the extension manager.
+	 *
+	 * @since 2.11.4
+	 * @param object $item
+	 * @return array
+	 */
+	private function get_item_data( $item ) {
+		return array(
+			'title'       => ! empty( $item->title ) ? $item->title : '',
+			'slug'        => ! empty( $item->slug ) ? $item->slug : '',
+			'image'       => ! empty( $item->image ) ? $item->image : '',
+			'description' => ! empty( $item->excerpt ) ? $item->excerpt : '',
+			'basename'    => ! empty( $item->custom_meta->basename ) ? $item->custom_meta->basename : '',
+			'tab'         => ! empty( $item->custom_meta->settings_tab ) ? $item->custom_meta->settings_tab : '',
+			'section'     => ! empty( $item->custom_meta->settings_section ) ? $item->custom_meta->settings_section : '',
+		);
+	}
+
+	/**
+	 * Gets the base url for the products remote request.
+	 *
+	 * @since 2.11.4
+	 * @return string
+	 */
+	private function get_products_url() {
+		if ( defined( 'EDD_PRODUCTS_URL' ) ) {
+			return EDD_PRODUCTS_URL;
+		}
+
+		return 'https://easydigitaldownloads.com/';
+	}
+
+	/**
+	 * Gets the default array for the body of the API request.
+	 * A class may override this by setting an array to query a tag or category.
+	 * Note that the first array key/value pair are used to create the option name.
+	 *
+	 * @since 2.11.4
+	 * @param int $item_id The product ID.
+	 * @return array
+	 */
+	private function get_api_body( $item_id ) {
+		return array(
+			'product' => $item_id,
+		);
+	}
+
+	/**
+	 * Gets the first key of an array.
+	 * (Shims array_key_first for PHP < 7.3)
+	 *
+	 * @since 2.11.4
+	 * @param array $array
+	 * @return string|null
+	 */
+	private function array_key_first( array $array ) {
+		if ( function_exists( 'array_key_first' ) ) {
+			return array_key_first( $array );
+		}
+		foreach ( $array as $key => $unused ) {
+			return $key;
+		}
+
+		return null;
+	}
+
+	/**
+	 * Checks whether a given option has "expired".
+	 *
+	 * @since 2.11.4
+	 * @param array|false $option
+	 * @return bool
+	 */
+	private function option_has_expired( $option ) {
+		return ! empty( $option['timeout'] ) && time() > $option['timeout'];
```

Comment:
```suggestion
		return empty( $option['timeout'] ) || time() > $option['timeout'];
```

Problem I'm having now is that this is returning `false` if `'timeout'` doesn't exist (my option is `false` due to not having nay data yet).

Reply:
I probably should update line 32 to be `if ( $option && ! $is_stale ) {` and the same for the all product data--that would have fixed this, too.